src/test/java/com/rocketden/main/api/ProblemTests.java

```diff
@@ -338,4 +342,40 @@ public void createProblemWithTestCasesSuccess() throws Exception {
         assertEquals(OUTPUT_2, case2.getOutput());
         assertFalse(case2.isHidden());
     }
+
+    @Test
+    public void getRandomProblemSuccess() throws Exception {
+        ProblemDto problem = createSingleProblem();
+
+        MvcResult result = this.mockMvc.perform(get(GET_PROBLEM_RANDOM)
+                .param(DIFFICULTY_KEY, "EASY"))
+                .andDo(print()).andExpect(status().isOk())
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+        ProblemDto actual = UtilityTestMethods.toObject(jsonResponse, ProblemDto.class);
+
+        assertEquals(problem.getName(), actual.getName());
+        assertEquals(problem.getDescription(), actual.getDescription());
+        assertEquals(ProblemDifficulty.EASY, actual.getDifficulty());
+        assertEquals(problem.getProblemId(), actual.getProblemId());
+        assertEquals(problem.getTestCases(), actual.getTestCases());
+    }
+
+    @Test
+    public void getRandomProblemNotFound() throws Exception {
```

Comment:
Can you add a third test where you add no problems to the database, and then test that you still get `ProblemError.NOT_FOUND`?

Reply:
This should be covered in the `getRandomProblemNotFound` test in `ProblemServiceTests` - lmk if you think I should add one here as well tho

Reply:
Yeah it's testing the same thing, so there's no real need (which in my mind again raises the question of why we have these two separate sets of tests). I suppose just in the future, if we remove one of the sets of testing, to ensure that this behavior is still tested.